ext/fetch/wasm_streaming.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_core::error::type_error;
+use deno_core::error::AnyError;
+use deno_core::v8;
+use deno_core::OpState;
+use deno_core::ResourceId;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+// The Wasm streaming compilation pipeline.
+pub fn handle_wasm_streaming(
+  state: Rc<RefCell<OpState>>,
+  scope: &mut v8::HandleScope,
+  value: v8::Local<v8::Value>,
+  mut wasm_streaming: v8::WasmStreaming,
+) {
+  let (url, rid) = match compile_response(scope, value) {
```

Comment:
I think we should make a call into JS here to do a branding check and extract the stream RID, as this could in theory be created w/a Response that is not sourced from fetch.